snapcraft/internal/pluginhandler/__init__.py

```diff
@@ -815,7 +815,7 @@ def _do_prime(self) -> None:
         snap_files, snap_dirs = self.migratable_fileset_for(steps.PRIME)
         _migrate_files(snap_files, snap_dirs, self.stagedir, self.primedir)
 
-        if self._snap_type == "app":
+        if self._snap_type == "app" and self._base:
```

Comment:
I don't know why, but lately I prefer `is not None` for explicit `None` checks.
This is fine though, just thought I'd mention it.

Reply:
Done